Implemented the tooltip for table column headers

nivzx opened the pull request:
**The PR fulfills these requirements:** (check all the apply)

- [x] It's submitted to the `main` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `feat: Add a button #xxx`, where "xxx" is the issue number).
- [x] When resolving a specific issue, the PR description includes `Closes #xxx`, where "xxx" is the issue number.
- [x] If changes were made to `ui` folder, unit tests (`make test`) still pass.
- [x] New/updated tests are included

Closes #1792

Added a optional `tooltip` parameter to `table_column`
```
ui.table(name='table', columns=[
            ui.table_column(name='col1', label='Column 1', tooltip="I am a tooltip"),
            ui.table_column(name='col2', label='Column 2'),
        ]
```

The column header with `tooltip` property is displayed with a tooltip icon and the tooltip content is displayed when it's hovered.
![image](https://github.com/h2oai/wave/assets/63404062/95a171a5-bf56-475f-af1f-41c2edd98013)

cc @mturoci

mturoci reviewed (changes requested):
Thanks @nivzx!

* The tooltip is too far away from the icon.
* Hovering over icon shouldn't bring up pointer cursor since clicking the info icon triggers no action.
* The table column styling should remain intact.
* Please add a unit tests to verify that the info button with tooltip is only rendered when `tooltip` prop is specified on a col. Columns without the prop should not render the info icon.

nivzx commented:
Fixed the styling issues. Will update the tests. Meanwhile please let me know your thoughts.
cc: @mturoci 

> [x] The tooltip is too far away from the icon.
> [x] Hovering over icon shouldn't bring up pointer cursor since clicking the info icon triggers no action.
> [x] The table column styling should remain intact.
> [] Please add a unit tests to verify that the info button with tooltip is only rendered when `tooltip` prop is specified on a col. Columns without the prop should not render the info icon.

<img width="584" alt="image" src="https://github.com/h2oai/wave/assets/63404062/452c5811-505c-44e1-8197-ceb61d57882a">

mturoci reviewed (changes requested):
The table styling should remain intact as mentioned previously.

![image](https://github.com/user-attachments/assets/26a13bf0-9e10-4baa-b2a3-530217ef1326)

You can use our visual regression testing tool:

* Checkout main
* `make build-ui`
* `cd tools/showcase && make visual-regression`
* chekout your feature branch
* `make build-ui`
* `cd tools/showcase && make visual-regression`
* `make test-result`.

See `Makefile` in `tools/showcase` for more info.

The tooltip is not centered, the icon is way too small and the spacing between other column icons is uneven.

![image](https://github.com/user-attachments/assets/20995a16-3991-46bc-9e4d-91853a91483d)

Also, the tooltip probably needs either a border or slightly different background so that it's easier to spot:

![image](https://github.com/user-attachments/assets/b6112496-21ef-4968-a1e4-ad0b899faa1a)

Maybe needed for the rest of the tooltips as well so prefer generic solution.

nivzx commented:
> The table styling should remain intact as mentioned previously.
> 
> 
> You can use our visual regression testing tool
> 
> The tooltip is not centered, the icon is way too small and the spacing between other column icons is uneven.
> 
> Also, the tooltip probably needs either a border or slightly different background so that it's easier to spot:
> 
> Maybe needed for the rest of the tooltips as well so prefer generic solution.

Got it! I'll try to refine this further. Thanks!

nivzx commented on ui/src/table.tsx:
```diff
@@ -433,18 +436,36 @@ const
       primaryColumnKey = m.columns.find(c => c.link)?.name || (m.columns[0].link === false ? undefined : m.columns[0].name),
       onRenderDetailsHeader = React.useCallback((props?: Fluent.IDetailsHeaderProps) => {
         if (!props) return <span />
+        const renderColumnHeaderTooltip = (tooltipHostProps: any) => {
```
Done

nivzx commented on ui/src/table.test.tsx:
```diff
@@ -1954,4 +1954,34 @@ describe('Table.tsx', () => {
+      const infoIcons = container.querySelectorAll("[data-icon-name='info-icon']") as NodeListOf<HTMLElement>
+      expect(infoIcons).toHaveLength(1)
+  
+      headers.forEach(header => {
```
Done

nivzx commented:
- Fixed the issues with header stylings and verified with the visual regression tool.
- Increased the visibility of the icon
- Fixed the misalignment of the tooltip with the icon
- Removed the unnecessary looping in `table.test.tsx`
- Properly typed `tooltipHostProps` with `IDetailsColumnRenderTooltipProps`

<img width="621" alt="image" src="https://github.com/user-attachments/assets/aacd3d3e-2485-4d53-b0d3-c588ad4ef528">

cc: @mturoci

swaldow commented:
Is there anything new about this feature? We think it's cool and would like to have it :)

mturoci commented:
@marek-mihok can you please review this PR and verify it works/didn't break anything once you have some time? Not urgent.
